drivers/sensor/iis2dh/iis2dh.c

```
@@ -0,0 +1,315 @@
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		ofs_start = ofs_stop = 1U;
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		ofs_start = ofs_stop = 2U;
+		break;
+	default:
+		ofs_start = 0U; ofs_stop = 2U;
+		break;
+	}
+
+	for (i = ofs_start; i <= ofs_stop ; i++) {
+		iis2dh_convert(pval++, iis2dh->acc[i], iis2dh->gain);
+	}
+}
+
+static int iis2dh_channel_get(struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		iis2dh_channel_get_acc(dev, chan, val);
+		return 0;
+	default:
+		LOG_DBG("Channel not supported");
+		break;
+	}
+
+	return -ENOTSUP;
+}
+
+static int iis2dh_config(struct device *dev, enum sensor_channel chan,
+			 enum sensor_attribute attr,
+			 const struct sensor_value *val)
+{
+	switch (attr) {
+#if (CONFIG_IIS2DH_RANGE == 0)
+	case SENSOR_ATTR_FULL_SCALE:
+		return iis2dh_set_range(dev, sensor_ms2_to_g(val));
+#endif
+#if (CONFIG_IIS2DH_ODR == 0)
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return iis2dh_set_odr(dev, val->val1);
+#endif
+	default:
+		LOG_DBG("Acc attribute not supported");
+		break;
+	}
+
+	return -ENOTSUP;
+}
+
+static int iis2dh_attr_set(struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		return iis2dh_config(dev, chan, attr, val);
+	default:
+		LOG_DBG("Attr not supported on %d channel", chan);
+		break;
+	}
+
+	return -ENOTSUP;
+}
+
+static int iis2dh_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct iis2dh_data *iis2dh = dev->driver_data;
+	union axis3bit16_t buf;
+
+	/* fetch raw data sample */
+	if (iis2dh_acceleration_raw_get(iis2dh->ctx, buf.u8bit) < 0) {
+		LOG_DBG("Failed to fetch raw data sample");
+		return -EIO;
```

Why not return the error you got from iis2dh_acceleration_raw_get()?